src/account.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+var Transaction = require('../src/transaction').Transaction
+
+function Account () {
+  this.transactions = []
+}
+
+Account.prototype.recordTransaction = function (amount, date) {
+  var transaction = new Transaction(amount, date);
+  this.transactions.push(transaction)
+};
+
+Account.prototype.madeDeposit = function (amount, date) {
+  validateDate(date);
+  this.recordTransaction(amount, date);
+};
+
+Account.prototype.withdrewMoney = function (amount, date) {
+  validateDate(date);
+  this.recordTransaction(-amount, date);
+};
+
+function validateDate (string) {
```

Comment:
Yes - definitely worth refactoring this.  Could you use the built-in `Date` constructor to do the hard work by rejecting any dates it can't parse?